torch/csrc/distributed/autograd/context/dist_autograd_context.cpp

```diff
@@ -94,10 +114,35 @@ void DistAutogradContext::setGraphTask(
 
 void DistAutogradContext::addOutstandingRpc(
     const std::shared_ptr<rpc::FutureMessage>& futureMessage) {
+  futureMessage->addCallback(std::bind(
+      &DistAutogradContext::outStandingRpcCallback,
+      this,
+      std::placeholders::_1));
   std::lock_guard<std::mutex> guard(lock_);
   outStandingRpcs_.push_back(futureMessage);
 }
 
+void DistAutogradContext::outStandingRpcCallback(const rpc::Message& message) {
+  if (message.type() == rpc::MessageType::EXCEPTION) {
+    // If we have an error, let the local autograd engine know about it.
+    std::string err(message.payload().begin(), message.payload().end());
+    auto exception = std::runtime_error(err);
+
+    // Enqueue 'ErrorFunc' on the local autograd engine.
```

Comment:
This seems like a really circuitous way to do this. Why not just expose a method on `graphTask_` that triggers an error? No need to create a synthetic task...

Reply:
I agree your proposal is a much simpler way to do this.